rd/triangle2D.h

```diff
@@ -1050,22 +1052,21 @@ class TRIANGLE{
                 
                 calculate_normals(X_MOD,Y_MOD);
 
-                return ;
 
         }
 
-        void calculate_normals(double X[3],double Y[3]){
+        void calculate_normals(double X_input[3],double Y_input[3]){
```

Comment:
X and Y will shadow the private members X, Y in the TRIANGLE class, use X_input and Y_input instead.